### PR TITLE
Only remove subtitles which do not encompass the seek target

### PR DESCRIPTION
**Pull request by bitmovin-kenny**
When seeking, we removed all active subtitles, even when we seeking inside the range of the active subtitle. In that case the UI would not display the Cue because the player will only fire a `CueExit` + `CueEnter` when we played outside of the Cue.

So in this PR we check the playback range of all active cues after a Seek / Timeshift operation has been completed and only remove cues which do not enclose the new current time.

### CODE REVIEW

File: src/ts/components/subtitleoverlay.ts [hunk 1]
```diff
@@ -121,11 +121,18 @@ export class SubtitleOverlay extends Container<ContainerConfig> {
       this.updateComponents();
     };
 
+    const clearInactiveSubtitles = () => {
+      const removedActiveSubtitles = subtitleManager.clearInactiveCues(player.getCurrentTime());
```
Comment (jmsn) [inline src/ts/components/subtitleoverlay.ts:125]: NIT: For clarity, I would choose a name that reflects that these are actually `cues`

File: src/ts/components/subtitleoverlay.ts [hunk 2]
```diff
@@ -470,6 +477,24 @@ class ActiveSubtitleManager {
     }
   }
 
+  /**
+   * Removes all active cues which don't enclose the given time
+   * @param time the time for which subtitles should remain
+   */
+  public clearInactiveCues(time: number): ActiveSubtitleCue[] {
+    const removedEvents: ActiveSubtitleCue[] = [];
```
Comment (jmsn) [inline src/ts/components/subtitleoverlay.ts:485]: NIT: Ar these cues or events? 😅

File: src/ts/components/subtitleoverlay.ts [hunk 3]
```diff
@@ -470,6 +477,24 @@ class ActiveSubtitleManager {
...
+    Object.keys(this.activeSubtitleCueMap).forEach(key => {
+      const activeCues = this.activeSubtitleCueMap[key];
+      activeCues.forEach(element => {
```
Comment (jmsn) [inline src/ts/components/subtitleoverlay.ts:488]: Also here use `cue`? 🙂

File: src/ts/components/subtitleoverlay.ts [hunk 4]
```diff
@@ -470,6 +477,24 @@ class ActiveSubtitleManager {
...
+        if (time < element.event.start || time > element.event.end) {
+          this.popCueFromMap(element.event);
+          removedEvents.push(element);
+        }
+      });
+    });
+    return removedEvents;
+  }
```
Comment (jmsn) [inline src/ts/components/subtitleoverlay.ts:496]: So we are preserving only the cue that lies at the given time. Can it happen that there are cues in the future (i.e. that lie after `time`) that should not be cleared? 🤔

**Review by jmsn — COMMENTED** [review]
LGTM

Just few NITs and a question 🙂

File: src/ts/components/subtitleoverlay.ts (hunk 2 above)
Comment (bitmovin-kenny) [reply src/ts/components/subtitleoverlay.ts:485]: It's actually a label and an event:
```ts
interface ActiveSubtitleCue {
  event: SubtitleCueEvent;
  label: SubtitleLabel;
}
```

File: src/ts/components/subtitleoverlay.ts (hunk 1 above)
Comment (bitmovin-kenny) [reply src/ts/components/subtitleoverlay.ts]: Done in https://github.com/bitmovin/bitmovin-player-ui/pull/573/commits/bf0e75d23a90bb38d251c8c6ece7b18e82d07737

File: src/ts/components/subtitleoverlay.ts (hunk 4 above)
Comment (bitmovin-kenny) [reply src/ts/components/subtitleoverlay.ts:496]: I can't think of a reason for there being active cues which should linger until they become active again. I would expect the player to emit `CueEnter` events when it reaches that range. Then we would create an active cue again. If we didn't clear it before, there would then be two cues maybe?

Comment (jmsn) [reply src/ts/components/subtitleoverlay.ts:496]: Thanks 👍 , I thought so, but wanted to ask if I understood it correctly. 🙂

**Review by jmsn — APPROVED** [review]

File: src/ts/components/subtitleoverlay.ts (hunk 2 above)
Comment (bitmovin-kenny) [reply src/ts/components/subtitleoverlay.ts]: renamed in https://github.com/bitmovin/bitmovin-player-ui/pull/573/commits/10c0774097ff2b736a84630fdbe233fdba8bc81c

**Pull request merged**